app/bod/management/commands/bod_migrate.py

```diff
@@ -0,0 +1,104 @@
+from bod.models import BodContactOrganisation
+from provider.models import Provider
+
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+
+class Command(BaseCommand):
+    help = "Migrates data from a BOD"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.messages = []
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--clear",
+            action="store_true",
+            help="Delete existing objects before importing",
+        )
+        parser.add_argument(
+            "--dry-run",
+            action="store_true",
+            help="Dry run, abort transaction in the end",
+        )
+        parser.add_argument(
+            "--verbose",
+            action="store_true",
+            help="Show more output",
+        )
+
+    def clear(self):
+        cleared, _ = Provider.objects.all().delete()
+        if cleared:
+            self.messages.append(f"{cleared} provider(s) cleared")
+
+    def import_providers(self, verbose):
+        added = 0
+        updated = 0
+        removed = 0
+        processed = set()
+
+        for organization in BodContactOrganisation.objects.all():
+            legacy_id = organization.pk_contactorganisation_id
+            processed.add(legacy_id)
+            provider, created = Provider.objects.get_or_create(_legacy_id=legacy_id)
+            changed = False
+            for provider_attribute, organization_attribute in (
+                ('name_de', 'name_de'),
+                ('name_de', 'name_de'),
+                ('name_en', 'name_en'),
+                ('name_fr', 'name_fr'),
+                ('name_it', 'name_it'),
+                ('name_rm', 'name_rm'),
+                ('acronym_de', 'abkuerzung_de'),
+                ('acronym_fr', 'abkuerzung_fr'),
+                ('acronym_en', 'abkuerzung_en'),
+                ('acronym_it', 'abkuerzung_it'),
+                ('acronym_rm', 'abkuerzung_rm'),
+            ):
+                old = getattr(provider, provider_attribute)
+                new = getattr(organization, organization_attribute)
+                if old != new:
+                    changed = True
+                    setattr(provider, provider_attribute, new)
+                    if verbose and not created:
+                        self.stdout.write(
```

Comment:
there are `self.print_warning`, `self.print_success` etc. (see https://github.com/geoadmin/service-stac/blob/d852020b9ed4ce64c454f188b7c1279569792fcd/app/stac_api/management/commands/remove_expired_items.py#L21) 
I'm unsure about which one is to be favoured but we should probably use the same everywhere and make sure the output is correctly formatted to be consumed by log parsing

Reply:
Great! I added the `CommandHandler` and `CustomBaseClass` to the code and added some tests.